Make README instructions useful

PeterJCLaw opened the pull request:
This now includes the instructions for building the python package on Ubuntu, which seems fairly representative of a Debian machine.

Note: the `debuild` doesn't actually work for me yet as I've not worked out how to get `python3-pygame` (which seems to be needed) installed; it's not in the repos for Ubuntu 16.04.

PeterJCLaw committed:
Make README instructions executable

PeterJCLaw committed:
Clarify build and install requirements

PeterJCLaw committed:
Drop pointless word

prophile commented:
The build dependencies are listed in `debian/control`: I think it might be better to recommend `mk-build-deps` rather than including a sample of the dependencies in the `README`.

PeterJCLaw committed:
Encourage use of mk-build-deps

This is better than duplicating a list of the dependencies in the README.

equivs is needed by mk-build-deps.

Adimote commented:
LGTM, BSEMA

nmalky commented on README.md:
```diff
@@ -17,16 +17,33 @@ Tour of the source
+Getting started
+---------------
+
+Since `robotd` vendors in April Tags for its vision support, which depends on
```
"vendors in" doesn't make much sense to me, and doesn't appear to be very commonly used in this way.
Does this mean that `robotd` includes April Tags when distributed? Or maybe that it depends on April Tags?

PeterJCLaw replied:
@nmalky it did at the time, though the AprilTags stuff has since been moved out to https://github.com/sourcebots/sb-vision.

The idea of vendoring something into another project (perhaps not the best terminology) is to include some dependency (usually recursively, often in binary-form though not necessarily) in a given project. This is usually done for developer convenience and/or to pin a version. It's not really an ideal thing to do, but is useful as a way to get started.

nmalky commented on README.md:
```diff
@@ -17,16 +17,33 @@ Tour of the source
+---------------
+
+Since `robotd` vendors in April Tags for its vision support, which depends on
+`libusb`, you'll need to install the development package for `libusb` in order
```
Is this correct? From https://github.com/sourcebots/sb-vision/blob/master/debian/control (no reference to libusb) and https://github.com/sourcebots/sb-vision/blob/807477ffdad87228be50475a8c84c1ecf94b325a/sb_vision/native/apriltag/contrib/april/README ("no external dependencies"), it seems like `libusb` isn't required for April Tags or the `sb-vision`.
(I haven't tried actually building/running, though)

PeterJCLaw replied:
I think the statement that AprilTags needs `libusb` is actually wrong. `robotd` depends on `libusb` directly.

PeterJCLaw replied:
I don't know whether we need `libusb`'s dev package any more.
